miio/airfresh_t2017.py

```diff
@@ -134,32 +162,60 @@ def control_speed(self) -> int:
     @property
     def dust_filter_life_remaining(self) -> int:
         """Remaining dust filter life in percent."""
-        return self.data["filter_intermediate"]
+        if (
+            "filter_intermediate" in self.data
+            and self.data["filter_intermediate"] is not None
+        ):
+            return self.data["filter_intermediate"]
```

Comment:
This could be simplified by simply checking for both values? Returns none if neither of the property is there, so the return value should be marked as Optional.
```
# filter_intermediate for model X, filter_rate for A1
return self.data.get("filter_intermediate", self.data.get("filter_rate"))
```